include/envoy/filesystem/filesystem.h

```diff
@@ -25,37 +30,55 @@ class File {
    *
    * @return bool whether the open succeeded
    */
-  virtual Api::SysCallBoolResult open() PURE;
+  Api::IoCallBoolResult open() {
+    if (isOpen()) {
+      return resultSuccess<bool>(true);
+    }
+
+    openFile();
+    return -1 != fd_ ? resultSuccess<bool>(true) : resultFailure<bool>(false, errno);
+  }
 
   /**
    * Write the buffer to the file. The file must be explicitly opened before writing.
    *
    * @return ssize_t number of bytes written, or -1 for failure
    */
-  virtual Api::SysCallSizeResult write(absl::string_view buffer) PURE;
+  Api::IoCallSizeResult write(absl::string_view buffer) {
+    const ssize_t rc = writeFile(buffer);
+    return -1 != rc ? resultSuccess<ssize_t>(rc) : resultFailure<ssize_t>(rc, errno);
```

Comment:
nit: prefer `rc != -1` here as it reads more naturally.

Reply:
Done.